container/container_unix.go

```diff
@@ -302,6 +302,9 @@ func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfi
 
 	// update HostConfig of container
 	if hostConfig.RestartPolicy.Name != "" {
+		if container.HostConfig.AutoRemove {
```

Comment:
Could we detect actual change in here. If container has `--restart=no` then update to `--restart=no` shouldn't be forbidden.